### ISSUE TITLE
Inset text

### ISSUE THREAD

**Issue by timpaul** [alphagov/govuk-design-system-backlog#136]
# New issue template

---

## What
Use inset text to draw attention to important content on the page.
![image](https://user-images.githubusercontent.com/1590604/39632055-e87ddb06-4fab-11e8-9d31-071d1cb8903c.png)

## Why
This style already exists in GOV.UK Elements, and is used on many services.

**Comment by stevenaproctor** [comment]
@timpaul I thought this component was deprecated. https://github.com/alphagov/govuk_elements/pull/586

Or at least a rethink of how inset text and warning text worked together.

**Comment by EmmaJem** [comment]
is there a reason why this component's border is sometimes blue? @timpaul

**Comment by vanitabarrett** [comment]
Hi @EmmaJem , have you got any examples of the component with a blue border?

**Comment by EmmaJem** [comment]
> Hi @EmmaJem , have you got any examples of the component with a blue border?

Hi here's and example from GOV.UK Passport replacement form: 


![MicrosoftTeams-image](https://user-images.githubusercontent.com/60506937/93472663-e2d80680-f8ec-11ea-83a1-9a13cd49161b.png)

**Comment by vanitabarrett** [comment]
Thanks for the example @EmmaJem . That looks like a custom component (or some overridden styles) added by a department, rather than something in the design system. The style and position of the blue version is a little strange, it looks a bit more like the "alert" or "notification banner" that other services have built and which the design system team are in the process of building. But the content is perhaps more suited to hint text on the input.

The inset text component in the design system is grey, and can be previewed here: https://design-system.service.gov.uk/components/inset-text/

**Comment by EmmaJem** [comment]
@vanitabarrett thank you very much for your help! I agree It was a little confusing and didn't seem to fit! 

Thank you :)

**Comment by edwardhorsford** [comment]
Quite a few services have alternate / additional inset styles - I do think there's a bit of an unmet need here.

**Comment by MhariMcNaught** [comment]
I'm looking at updating the styling of our inset text component using one of the colours from our colour palette. 

The grey in the GDS inset text component fails the contrast checker for graphical objects and user Interface components, using https://webaim.org/resources/contrastchecker/ 

Do the GDS team think this does not matter for the inset text component?

**Comment by querkmachine** [comment]
Hi @MhariMcNaught,

We don't consider the contrast of the border to be an issue in this context, as we don't consider it to be a 'user interface component' nor a 'graphical object' necessary to understand the content, which is what [WCAG criterion 1.4.11](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) applies to. 

Additionally, if a user is unable to clearly see the border due to contrast issues, the content still appears visually inset. 

Hope that helps!

**Comment by Graeme-Laurenson** [comment]
Hi
We've developed the component a little to show as below .. this is to help the user identify both the Employer and Employment period start date in one easy to scan ..
Popping it here for initial thoughts before proceeding .. 
Fire away!
G.
![Screenshot 2022-08-11 at 16 42 53](https://user-images.githubusercontent.com/85180662/184174672-47ed1c52-c4fd-400b-8f04-6ac8ecd28a75.png)

**Comment by beccagorton-1** [comment]
Not sure if this is the right issue, but we have a slightly more unique service. For our DFE service, which is a training platform, we needed a component that highlighted actionable takeaways; something that encourages users to stop and reflect. We initially used the inset text standard component, and users fed back that it felt too dull. 

We designed an adaptation of the inset text component with colour to help this content stand out more. For users who may see in greyscale, we have assumptions that the solid grey block would stand out more than the subtle grey left border. 

We chose the teal colour as this was empathy tested using colour filters and appeared largely the same. Equally, we already used green and blue feedback boxes for other messages. 

This component will be tested through an accessibility audit soon. 
<img width="819" alt="Screenshot 2022-05-17 at 10 33 59" src="https://user-images.githubusercontent.com/104020160/184670943-a916db05-ed75-4d1b-b577-beebea71aafe.png">

**Comment by cjforms** [comment]
@beccagorton-1 
If you're open to another suggestion, there's another way to grab and focus attention from the reader (in this case, the practitioner) which is to make the reflection questions into actual questions with a space for them to reply with their answers. 

Having a 'box to type into' is a very compelling feature on the page - my eyetracking studies (years ago when we still did that sort of thing, admittedly) showed people's eyes going directly to the boxes. 

(An aside: The only thing that can compete as an attractive feature on a page is the eyes of a person in a photograph looking out at the reader, but I'm not recommending that in this context due to the difficulties of representation here). 

There is an another advantage of giving people boxes to type into: the box is a standard component in the design system.

You might very reasonably point out that you'd have to think about what to do with the answers that people typed in. True. To which I'd ask: What do you expect them to do with the answers to your prompts at the moment?

**Comment by beccagorton-1** [comment]
Hi @cjforms! I'm really glad you mentioned it, as this is exactly what we're testing right now :) 
We're trying to refine what that looks like, and I will definitely update once we've finished iterating it. 

In terms of the prompt, there will be situations where the box might not be quite right (we're currently testing variations of content within the prompts as well) and we've learned from users that they don't want the box on every page, but are hapy to have prompts on every page. We've also found that the prompts make the course materials a little more engaging, as previously it was all body text with the original inset text component appearing sparingly.

**Comment by selfthinker** [comment]
The govuk publishing components have a component called [intervention](https://components.publishing.service.gov.uk/component-guide/intervention). That looks like it derived from the inset text component and changed some colours.

<img width="921" alt="Intervention component with a green border on the left and a grey background" src="https://github.com/alphagov/govuk-design-system-backlog/assets/108893/edf8df10-7c0b-4cf6-9297-3e327d55010b">

**Comment by nick-wall** [comment]
We've recently used a version of this in a service for recruiters to check a teacher's record. One the main things they're looking out for is if the teacher has any 'restrictions' such as a misconduct hearing that has prohibited them from teaching.

They won't often see them, but if there is one, we use the inset text component in red. 

![image](https://github.com/alphagov/govuk-design-system-backlog/assets/149669706/55429d15-9ee1-4aaf-920a-a8a723afb116)

We didn't use [warning text](https://design-system.service.gov.uk/components/warning-text/?) because it isn't always a legal warning, depending on the education setting and restriction type, and it doesn't lend itself well to multi-line content. Ours has a heading and some body content.

We didn't use a [notification banner](https://design-system.service.gov.uk/components/notification-banner/?) because this is more typical for a service update or flash message. It was also quite large and bordered, we wanted to start with something simpler.

We used red because it's an alert, we don't use this colour for anything else and it's familiar to users from something similar in the legacy service. It's worked with a couple of people with deuteranopia (red-green) because there are no other prominent components or colours nearby, and we've made an assumption that for greyscale viewers it would appear similar to the component's standard state.

It's tested very well in a couple of rounds of research. Every participant noticed it. Nearly every participant:
- noticed it before anything else
- mostly only talked about this

Quotes included:
> "the red bar stands out immediately, it jumped out and hit me in the face"

> "red bar is good it immediately jumps out, there is an issue with this person"​

> ""it's really clear she is on prohibition, failed induction and contact information for TRA very clear. Colouring is very useful and I am very visual"

We talked about this in a service assessment and was assessed as met the standards. It's gone into public beta, so we'll be interested to see if any more feedback comes up.

**Comment by joelanman** [comment]
@nick-wall thanks for feeding that back, one point - it should probably use the standard govuk red, at a glance it looks different:

https://design-system.service.gov.uk/styles/colour/

**Comment by nick-wall** [comment]
> @nick-wall thanks for feeding that back, one point - it should probably use the standard govuk red, at a glance it looks different:
> 
> https://design-system.service.gov.uk/styles/colour/

Good point - the screenshot is using 'govuk-red' in the Figma design system, but is actually slightly darker in production

![image](https://github.com/alphagov/govuk-design-system-backlog/assets/149669706/3a27626d-73c8-4931-bcee-5efac35019df)

**Comment by TKDev7** [comment]
The component page says that I can use this component for quotes. When doing so, do I need to add `role=blockquote`?  Or maybe instead use a `<blockquote>` instead of a `<div>` as shown in the example?

**Comment by rajyog7997** [comment]
How many insert text can we use on the page?
